File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.js
function solution(sequence, k) {
    var answer = new Array(2);
    
    let queue = [];
    let len = Number.MAX_SAFE_INTEGER;
    let sum = 0;
    let idx = 0;
    for (let i = 0; i < sequence.length; i++) {
        
        let num = sequence[i];
        queue.push(num);
        sum += num;
        
        while (sum > k) {
            sum -= queue[idx++];
        }
        
        if (sum === k) {
            if (len > i - idx) {
                answer[0] = idx;
                answer[1] = i;
                len = i - idx;
            }
        }
    }
    return answer;
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.java
import java.util.*;

class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        Queue<Integer> queue = new ArrayDeque<>();
        
        int sum = 0;
        int idx = 0;
        int len = Integer.MAX_VALUE;
        for (int i = 0; i < sequence.length; i++) {
            
            int num = sequence[i];
            queue.offer(num);
            sum += num;
            
            while (sum > k) {
                sum -= queue.poll();
                idx++;
            }
            
            if (sum == k) {
                if (len > i - idx) {
                    answer[0] = idx;
                    answer[1] = i;
                    len = i - idx;
                }
            }
        }
        return answer;
    }
}